viewer/packages/camera-manager/src/ComboControls.ts

```diff
@@ -13,54 +13,95 @@ import {
   //================================================
   // INSTANCE FIELDS
   //================================================
 
   public dispose: () => void;
 
+  private _firstPersonMode = false;
   private _enabled: boolean = true;
   private _options: ComboControlsOptions = CreateDefaultControlsOptions();
 
-  private _temporarilyDisableDamping: boolean = false;
   private readonly _domElement: HTMLElement;
   private readonly _camera: PerspectiveCamera | OrthographicCamera;
-  private readonly _reusableCamera: PerspectiveCamera | OrthographicCamera;
 
+  // First persion mode: Camera is located in center of this sphere with Target on the surface
+  // Orbit mode        : Camera is located on the surface of the sphere with Target in center
+  // Se figure below
   private readonly _target: Vector3 = new Vector3();
   private readonly _targetEnd: Vector3 = new Vector3();
-  private readonly _spherical: Spherical = new Spherical();
-  private readonly _sphericalEnd: Spherical = new Spherical();
-  private readonly _scrollTarget: Vector3 = new Vector3();
-  private readonly _viewTarget: Vector3 = new Vector3();
+  private readonly _cameraVector: Spherical = new Spherical();
+  private readonly _cameraVectorEnd: Spherical = new Spherical();
+
+  // The Translation are used only if first person mode is enabled, it translates the camera
+  // and the lookAt without changing the target or cameraVector.
+  private readonly _translation: Vector3 = new Vector3();
+  private readonly _translationEnd: Vector3 = new Vector3();
+
+  private readonly _scrollTarget: Vector3 = new Vector3(); // When using the wheel this is the maxiumum point to scroll to when mouseWheelAction === 'zoomToCursor'
+  private readonly _viewTarget: Vector3 = new Vector3(); // used as target when _options.lookAtViewTarget is true
 
   private readonly _rawCameraRotation = new Quaternion();
-  private readonly _accumulatedMouseMove: Vector2 = new Vector2();
+  private readonly _accumulatedMouseRotation: Vector2 = new Vector2();
   private readonly _keyboard: Keyboard;
   private readonly _pointEventCache: Array<PointerEvent> = [];
 
   // Temporary objects used for calculations to avoid allocations
   private readonly _reusableVector3s = new ReusableVector3s();
-  private readonly _raycaster: Raycaster = new Raycaster();
+
+  //  The camera always lookAt the Target
+  //
+  //            Orbit Mode
+  //          , - ~ ~ ~ - ,
+  //      , '               ' ,
+  //    ,                       ,       Formula used in both modes:
+  //   ,                         ,      CameraVector = CameraPosition - Target
+  //  ,          Target           ,     Target = CameraPosition - CameraVector
+  //  ,             *             ,     CameraPosition = Target + CameraVector
+  //  ,              \            ,
+  //   ,  CameraVector\          ,
+  //    ,              \        ,
+  //      ,             \    , '
+  //        ' - , _  _ , # <------ CameraPosition
```

Comment:
How did you create these, @nilscognite? Manually?

Reply:
Pure art talent!

Reply:
No. Someone else did it for me at https://ascii.co.uk/art